Sources/Brave/Frontend/Sync/BraveCore/BraveSyncAPIExtensions.swift

```diff
@@ -30,6 +30,10 @@ extension BraveSyncAPI {
   var isInSyncGroup: Bool {
     return Preferences.Chromium.syncEnabled.value
   }
+  
+  var shouldLeaveSyncGroup: Bool {
+    return (isInSyncGroup && !isSyncFeatureActive && !isFirstSetupComplete) || (isInSyncGroup && isSyncAccountDeletedNoticePending)
```

Comment:
This is sort of hard to read, maybe we can simplify this or expand it a bit?

```swift
var shouldLeaveSyncGroup: Bool {
  if !isInSyncGroup { 
    return false
  }
  return !(isSyncFeatureActive && isFirstSetupComplete) || isSyncAccountDeletedNoticePending
}
```

Reply:
Fixed at https://github.com/brave/brave-ios/pull/7031/commits/3d172a29f0a8d62dacb9675594e1f0f30ae29ee7, though using guard might be even better in this case